ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ManagementActor.scala

```diff
@@ -120,11 +120,27 @@ class ManagementActor(managers: Map[ProcessingType, ProcessManager],
   //TODO: In future we should move this functionality to ProcessManager.
   private def handleObsoleteStatus(processStateDefinitionManager: ProcessStateDefinitionManager, processState: Option[ProcessState], lastAction: Option[ProcessAction]): ProcessStatus =
     (processState, lastAction) match {
-      case (Some(state), _)  => ProcessStatus.create(state, lastAction)
-      case (None, Some(action)) if action.isCanceled => ProcessStatus(SimpleStateStatus.Canceled, processStateDefinitionManager)
-      case (None, None) => ProcessStatus(SimpleStateStatus.NotDeployed, processStateDefinitionManager)
+      case (Some(state), _)  => handleMismatchStateVersion(state, lastAction)
+      case (None, Some(act)) if act.isCanceled => ProcessStatus.simple(SimpleStateStatus.Canceled)
+      case (None, Some(act)) if act.isDeployed => ProcessStatus.simpleErrorShouldRunning(act.processVersionId, act.user, Option.empty)
+      case (None, None) => ProcessStatus.simple(SimpleStateStatus.NotDeployed)
     }
 
+  //This method handles some corner cases for mismatch states version
+  //TODO: In future we should move this functionality to ProcessManager.
+  private def handleMismatchStateVersion(processState: ProcessState, lastAction: Option[ProcessAction]): ProcessStatus = {
```

Comment:
I still see here a lot not matched strange situations:
1. `case (None, _)` - AFAIK it is when old process has missing version in state - we should return `ProcessStatus(processState)` + add some comment
2. `case (Some(_), Some(act)) if act.isCanceled && stateVersion.versionId != action.processVersionId` - what will be then? 
Currently will throw `MatchError`. I wonder if we could handle situation that we already know and if unknown, strange situation occurs, return some unknown state and log it as a warning. WDYT?

Reply:
Good catches!
1. think it's not possible anymore - but if so we can probably remove Option from version..
2. I think in this method we can just ignore actions other than deploy (filter) and case Some(version), None should apply
3. And maybe we should return "generic" error in other cases?

Reply:
1, 3 looks ok for me
Regards to 2. We can do it in this method (then we should also change method name to `handleMismatchDeployedStateVersion`), but on upper level (`handleObsoleteStatus`) we should handle `case (Some(_), Some(act)) if act.isCanceled`. It can be done be generalization existing case to `case (_, Some(act)) if act.isCanceled` or adding another one. Otherwise we will got MatchError there....